extensions/qcom/GLSL_QCOM_image_processing.txt

```diff
@@ -0,0 +1,231 @@
+
+
+Modifications to the OpenGL ES Shading Language Specification, Version 3.10, dated 29 January 2016
+
+    Including the following line in a shader can be used to control the
+    language features described in this extension:
+
+        #extension GL_QCOM_image_processing   : <behavior>
+
+    where <behavior> is as specified in Section 3.3.
+
+    New preprocessor #defines are added to the OpenGL Shading Language:
+
+        #define GL_QCOM_image_processing     1
+
+    Add to the table at the end of Subsection 8.9.1 the following image processing functions. Note
+	that these built-ins rely on separate samplers and textures as defined in GL_KHR_vulkan_glsl.
+
+    +------------------------------------+--------------------------------------------+
+    | Syntax                             | Description                                |
+    +------------------------------------+--------------------------------------------+
+    |   vec4 textureWeightedQCOM(        | weighted sample operation multiplies       |
+    |       sampler2D tex,               | a 2D kernel of filter weights with a corr- |
+    |       vec2      P,                 | esponding region of sampled texels and     |
+    |       sampler2DArray weight)       | sums the results to produce the output     |
+    |                                    | value.                                     |
+    +------------------------------------+--------------------------------------------+
+    |   vec4 textureBoxFilterQCOM(       | Linear operation taking average of pixels  |
+    |       sampler2D tex,               | within the spatial region described by     |
+    |       vec2      P,                 | boxSize.  The box is centered at coordinate|
+    |       vec2      boxSize)           | P and has width and height of boxSize.x    |
+    |                                    | and boxSize.y.                             |
+    +------------------------------------+--------------------------------------------+
+    |   vec4 textureBlockMatchSADQCOM(   | Block matching operation measures the      |
+    |       sampler2D target             | correlation (or similarity) of the target  |
+    |       uvec2     targetCoord,       | block and reference block.  targetCoord    |
+    |       sampler2D reference,         | and refCoord specify the bottom-left corner|
+    |       uvec2     refCoord,          | of the block in target and reference       |
+    |       uvec2     blockSize)         | images. The error metric is the Sum of     |
+    |                                    | Absolute Differences (SAD).                |
+    +------------------------------------+--------------------------------------------+
+    |   vec4 textureBlockMatchSSDQCOM(   | Block matching operation measures the      |
+    |       sampler2D target             | correlation (or similarity) of the target  |
+    |       uvec2     targetCoord,       | block and reference block.  TargetCoord    |
```

Comment:
There's not much definition of what the return value might be here. Is there anything more you can say to give a clearer idea of how it should be used? Or a reference for where it could be found? This doesn't mention, for example, the ranges of values, whether higher or lower means more similar, what the different vector components mean.

Reply:
Updated with a little more information and a reference to the Vukan spec where the details are specified.